### PR TITLE
Update Initial Sale contract

### PR DESCRIPTION
**Pull request by nimbus-devs**

### CODE REVIEW

File: src/NimbusInitialSale.sol [hunk 1]
```diff
@@ -151,9 +155,8 @@ contract NimbusInitialSale is Ownable, Pausable {
     INimbusStakingPool[] public stakingPools;
     address public recipient;                      
    
-    uint public ethNbuExchangeRate;
-    mapping(address => uint) public tokenNbuExchangeRates;
-
+    INimbusRouter public swapRouter;
```
Comment (PedroReyes) [inline src/NimbusInitialSale.sol:158]: What about changing the name INimbusRouter for INimbusSwapRouter?

File: src/NimbusInitialSale.sol [hunk 2]
```diff
@@ -267,28 +277,28 @@ contract NimbusInitialSale is Ownable, Pausable {
     }
     
     function buyExactNbuForTokens(address token, uint nbuAmount, address nbuRecipient) external whenNotPaused {
-        require(tokenNbuExchangeRates[token] > 0, "Not initialized token");
+        require(allowedTokens[token], "Not allowed token");
```
Comment (PedroReyes) [inline src/NimbusInitialSale.sol:280]: Nice improvement! ✌. More clear now!

File: src/NimbusInitialSale.sol [hunk 3]
```diff
@@ -120,7 +120,7 @@ interface INBU {
     function balanceOf(address account) external view returns (uint256);
     function transfer(address recipient, uint256 amount) external returns (bool);
     function approve(address spender, uint256 amount) external returns (bool);
-    function give(address recipient, uint256 amount) external;
+    function give(address recipient, uint256 amount) external returns (bool);
```
Comment (PedroReyes) [inline src/NimbusInitialSale.sol:123]: This change has not been reflected in the NBU.sol file, no? If so, that would mean the re-deployment of NBU and the re-emission of these tokens which have already been given to phase 1 users

Comment (lfons) [reply src/NimbusInitialSale.sol:123]: Good catch Pedro! @nimbus-devs, can we get any updates on this one? Thanks

**Pull request merged**